library/Icingadb/Web/Controller.php

```diff
@@ -550,6 +550,86 @@ public function export(Query ...$queries)
             return true;
         }
 
+        if ($this->format === 'json' || $this->format === 'csv') {
+            $isJsonFormat = $this->format === 'json';
+            $response = $this->getResponse();
+
+            $fileName = $this->getRequest()->getControllerName();
+            if ($isJsonFormat) {
+                $response
+                    ->setHeader('Content-Type', 'application/json')
+                    ->setHeader('Cache-Control', 'no-store')
+                    ->setHeader(
+                        'Content-Disposition',
+                        'inline; filename=' . $fileName . '.json'
+                    )
+                    ->sendResponse();
+            } else {
+                $response
+                    ->setHeader('Content-Type', 'text/csv')
+                    ->setHeader('Cache-Control', 'no-store')
+                    ->setHeader(
+                        'Content-Disposition',
+                        'attachment; filename=' . $fileName . '.csv'
+                    )
+                    ->sendResponse();
+            }
+            // Disable output buffering to stream large amounts of data without a memory exhausted error.
+            ob_end_clean();
+            set_time_limit(0);
+
+            $query = $queries[0];
+            $query->limit(Url::fromRequest()->getParam('limit'));
+            $query->resetOffset();
+            $this->applyRestrictions($query);
+
+            $keyName = is_array($query->getModel()->getKeyName())
+                ? $query->getModel()->getKeyName()
+                : (array) $query->getModel()->getKeyName();
+            $col = array_merge($keyName, $query->getModel()->getColumns());
+
+            if ($isJsonFormat) {
+                echo('[');
+            }
+
+            $rs = $query->execute()->disableCache();
+
+            foreach ($rs as $count => $row) {
+                if ($count) {
+                    $separator = $isJsonFormat ? ',' : PHP_EOL;
+                    echo($separator);
+                }
+
+                foreach ($col as $alias => $name) {
+                    if (is_int($alias)) {
+                        $alias = $name;
+                    }
+
+                    $result[$alias] = $row->$alias;
+                    if (
+                        $alias === 'id' || substr($alias, -3) === '_id'
+                        || substr($alias, -9) === '_checksum'  || substr($alias, -4) === '_bin'
+                    ) {
+                        $result[$alias] = base64_encode($row->$alias);
+                    }
+                }
+
+                if ($count === 0 && ! $isJsonFormat) {
+                    echo('SEP=,' . "\r\n");
```

Comment:
Where does this come from? It confuses my libre office calc:
![Screenshot from 2021-09-24 14-36-32](https://user-images.githubusercontent.com/16668527/134675238-276b4c07-7dd6-4384-85d7-11a8e2d940ca.png)

Reply:
This tells Excel how to separate the columns. I added this because otherwise Excel inserts all the columns into the first cell.

Reply:
I have tested the `csv` in the monitoring module. My Excel also inserts the columns there in the first cell. (so this is something with `Excel on Mac`, not the export code). Therefore, we do not need this.